packages/reports/src/reportBuilder/fetch/functions/dataGroups.ts

```diff
@@ -0,0 +1,57 @@
+import { Aggregator } from '../../../aggregator';
+import { FetchReportQuery } from '../../../types';
+import { FetchResponse } from '../types';
+import { Row } from '../../types';
+
+type DataGroupsFetchParams = {
+  programCode: string;
+};
+
+const fetchEvents = async (
+  aggregator: Aggregator,
+  query: FetchReportQuery,
+  params: DataGroupsFetchParams,
+): Promise<FetchResponse> => {
+  const { programCode } = params;
+  const { organisationUnitCode, startDate, endDate } = query;
+  const response = (await aggregator.fetchEvents(
+    programCode,
+    {
+      useDeprecatedApi: false,
+      dataServices: { isDataRegional: true },
+      organisationUnitCodes: [organisationUnitCode],
+      startDate,
+      endDate,
+    },
+    {},
+  )) as Row[];
```

Comment:
That's a good idea. I think after we deliver PSSS it will be a good time for us to find the right place to define common types between Tupaia packages (Analytic, Event, AnalyticCluster, etc.)